pkg/config/config_test.go

```diff
@@ -247,8 +295,52 @@ func TestLoadProxyEnvAndConf(t *testing.T) {
 	proxies := GetProxies()
 	assert.Equal(t,
 		&Proxy{
-			HTTP:    "http_conf",
+			HTTP:    "http_env",
 			HTTPS:   "",
 			NoProxy: []string{"d", "e", "f"}},
 		proxies)
 }
+
+func TestLoadProxyDDSpecificEnvAndConf(t *testing.T) {
+	os.Setenv("DD_PROXY_HTTP", "http_env")
+	Datadog.Set("proxy.no_proxy", []string{"d", "e", "f"})
+	Datadog.Set("proxy.http", "http_conf")
+	defer os.Unsetenv("DD_PROXY_HTTP")
+	defer Datadog.Set("proxy", nil)
+
+	loadProxyFromEnv()
+	proxies := GetProxies()
+	assert.Equal(t,
+		&Proxy{
+			HTTP:    "http_env",
+			HTTPS:   "",
+			NoProxy: []string{"d", "e", "f"}},
+		proxies)
+}
+
+func TestLoadProxyEmptyValuePrecedence(t *testing.T) {
+	os.Setenv("DD_PROXY_HTTP", "")
+	os.Setenv("DD_PROXY_NO_PROXY", "a b c")
+	os.Setenv("HTTP_PROXY", "env_http_url")
+	os.Setenv("HTTPS_PROXY", "")
+	os.Setenv("NO_PROXY", "")
```

Comment:
Could we add a test on `NO_PROXY` since the split is different from `DD_NO_PROXY` (space vs coma).

Reply:
other tests cover this, but I can create a specific test just for this yes

Reply:
actually I think this is covered well-enough by the `TestLoadProxyStdEnvOnly` and `TestLoadProxyDDSpecificEnvOnly` tests, I added a few comments there to make it more obvious (02be916)